macros/plugins/spark/create_external_table.sql

```diff
@@ -2,17 +2,25 @@
 
     {%- set columns = source_node.columns.values() -%}
     {%- set external = source_node.external -%}
-    {%- set partitions = external.partition -%}
+    {%- set partitions = external.partitions -%}
     {%- set options = external.options -%}
 
 {# https://spark.apache.org/docs/latest/sql-data-sources-hive-tables.html #}
     create table {{source(source_node.source_name, source_node.name)}} 
     {%- if columns|length > 0 %} (
         {% for column in columns %}
             {{column.name}} {{column.data_type}}
-            {{- ',' if not loop.last -}}
+            {{- ',' if not loop.last or partitions is not none -}}
```

Comment:
I think that all the changes from here to the end of the macro is not needed. See [this pr](https://github.com/dbt-labs/dbt-external-tables/pull/116) for more info. Would be good do verify this through some testing.

Reply:
They are needed @JCZuurmond. Look at this issue that I created https://github.com/dbt-labs/dbt-external-tables/issues/108 
The problem is that also the format of the macro is wrong, so if you just fix the typo, with the current macro it will absolutely fail since it will add the data types where the partitions are defined instead of where the columns are defined. Moreover, the partitions need to be declared as columns!

Reply:
I think that the columns should be defined under `columns` OR under `partitions. They will all end up as columns, but some are partitioned columns.